exotica/src/Scene.cpp

```diff
@@ -380,6 +444,41 @@ namespace exotica
       return attached_objects_.find(name)!=attached_objects_.end();
   }
 
+  void Scene::addTrajectoryFromFile(const std::string& link, const std::string& traj)
+  {
+      addTrajectory(link, loadFile(traj));
+  }
+
+  void Scene::addTrajectory(const std::string& link, const std::string& traj)
+  {
+      addTrajectory(link, std::shared_ptr<Trajectory>(new Trajectory(traj)));
+  }
+
+  void Scene::addTrajectory(const std::string& link, std::shared_ptr<Trajectory> traj)
+  {
+      const auto& tree = kinematica_.getTreeMap();
+      const auto& it = tree.find(link);
+      if(it==tree.end()) throw_pretty("Can't find link '"<<link<<"'!");
+      if(traj->getDuration()==0.0) throw_pretty("The trajectory is empty!");
+      trajectory_generators_[link] = std::pair<std::shared_ptr<KinematicElement>, std::shared_ptr<Trajectory>>(it->second, traj);
+      it->second->IsTrajectoryGenerated = true;
+  }
+
+  std::shared_ptr<Trajectory> Scene::getTrajectory(const std::string& link)
+  {
+      const auto& it = trajectory_generators_.find(link);
+      if(it==trajectory_generators_.end()) throw_pretty("No trajectory genrator defined for link '"<<link<<"'!");
+      return it->second.second;
+  }
+
+  void Scene::removeTrajectory(const std::string& link)
+  {
+      const auto& it = trajectory_generators_.find(link);
+      if(it==trajectory_generators_.end()) throw_pretty("No trajectory genrator defined for link '"<<link<<"'!");
+      it->second.first->IsTrajectoryGenerated = false;
+      trajectory_generators_.erase(it);
```

Comment:
should this also delete the link?

Reply:
Could do. It will witch to the original KDL::Segment offset after this in the current implementation. I don't expect this to be used at all though. The scene is supposed to be set up once and stay that way until its destroyed. I added removing just to keep the trajectories editable.